scripts/src/javatests/org/oppia/android/scripts/coverage/RunCoverageTest.kt

```diff
@@ -0,0 +1,435 @@
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun tearDown() {
+    System.setOut(originalOut)
+    scriptBgDispatcher.close()
+  }
+
+  @Test
+  fun testRunCoverage_testFileExempted_noCoverage() {
+    val exemptedFilePath = "app/src/main/java/org/oppia/android/app/activity/ActivityComponent.kt"
+
+    RunCoverage(
+      "${tempFolder.root}",
+      exemptedFilePath,
+      commandExecutor,
+      scriptBgDispatcher
+    ).execute()
+
+    assertThat(outContent.toString())
+      .isEqualTo("This file is exempted from having a test file. Hence No coverage!\n")
+  }
+
+  @Test
+  fun testRunCoverage_validSampleTestFile_returnsCoverageData() {
+    testBazelWorkspace.initEmptyWorkspace()
+
+    val sourceContent =
+      """
+      package com.example
+      
+      class TwoSum {
+      
+          companion object {
+              fun sumNumbers(a: Int, b: Int): Any {
+                  return if (a ==0 && b == 0) {
+                      "Both numbers are zero"
+                  } else {
+                      a + b
+                  }
+              }
+          }
+      }
+      """.trimIndent()
+
+    val testContent =
+      """
+      package com.example
+      
+      import org.junit.Assert.assertEquals
+      import org.junit.Test
+      
+      class TwoSumTest {
+      
+          @Test
+          fun testSumNumbers() {
+              assertEquals(TwoSum.sumNumbers(0, 1), 1)
+              assertEquals(TwoSum.sumNumbers(3, 4), 7)         
+              assertEquals(TwoSum.sumNumbers(0, 0), "Both numbers are zero")
+          }
+      }
+      """.trimIndent()
+
+    testBazelWorkspace.addSourceAndTestFileWithContent(
+      filename = "TwoSum",
+      sourceContent = sourceContent,
+      testContent = testContent,
+      sourceSubpackage = "coverage/main/java/com/example",
+      testSubpackage = "coverage/test/java/com/example"
+    )
+
+    val result = RunCoverage(
+      "${tempFolder.root}",
+      "coverage/main/java/com/example/TwoSum.kt",
+      longCommandExecutor,
+      scriptBgDispatcher
+    ).execute()
+
+    val expectedResultList = mutableListOf(
+      listOf(
+        "SF:coverage/main/java/com/example/TwoSum.kt",
+        "FN:7,com/example/TwoSum${'$'}Companion::sumNumbers (II)Ljava/lang/Object;",
+        "FN:3,com/example/TwoSum::<init> ()V",
+        "FNDA:1,com/example/TwoSum${'$'}Companion::sumNumbers (II)Ljava/lang/Object;",
+        "FNDA:0,com/example/TwoSum::<init> ()V",
+        "FNF:2",
+        "FNH:1",
+        "BRDA:7,0,0,1",
+        "BRDA:7,0,1,1",
+        "BRDA:7,0,2,1",
+        "BRDA:7,0,3,1",
+        "BRF:4",
+        "BRH:4",
+        "DA:3,0",
+        "DA:7,1",
+        "DA:8,1",
+        "DA:10,1",
+        "LH:3",
+        "LF:4",
+        "end_of_record"
+      )
+    )
+
+    assertThat(result).isEqualTo(expectedResultList)
+  }
+
+  @Test
+  fun testRunCoverage_validScriptPathSampleTestFile_returnsCoverageData() {
```

Comment:
I'm not sure that I fully understand what is meant by this condition (valid script path sample test file) vs. the one above or tests below. Could you maybe clarify in the test names exactly the differences between the tests?

Reply:
The test cases are for scenarios to test for various modules
1. script module needs to have the `javatests/...`
2. app module have `test/.../___Test.kt `or `test/.../___LocalTest.kt` or `sharedTest/...`
3. default behaviour should correspond to `test/...`

And this particular `testRunCoverage_validScriptPathSampleTestFile_returnsCoverageData` test case checks the scenario if a file provided from the scripts module correctly maps its test file and Bazel targets and executes them. Same with others for respective app, default behaviours

Reply:
The renamed methods are much clearer--thanks!